### PR TITLE
ghcup: 0.1.19.2 and update livecheck

### PR DESCRIPTION
**Pull request by darklore**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
- Modify sha256 of 0.1.19.2 release
  - v0.1.19.2 is released at Feb 24. https://github.com/haskell/ghcup-hs/releases/tag/v0.1.19.2
  - I don't know why this formula could use 0.1.19.2 before the release.
- Use github latest release for version check to avoid using unreleased git tag.
- Use ghc formula instead of ghc@9.2 because it seems allows Cabal 3.8
- Add test to check version number.

### CODE REVIEW

**Comment by chenrui333** [comment]
> * I don't know why this formula could use 0.1.19.2 before the release.

TLDR is upstream did a re-tag for 0.1.19.2, it was clear for us when we were trying to ship 0.21 tag
- https://github.com/Homebrew/homebrew-core/pull/120608
- https://github.com/haskell/ghcup-hs/issues/749

I think the livecheck update makes sense.

**Review by chenrui333 — DISMISSED** [review]

**Review by samford — COMMENTED** [review]
Using `GithubLatest` makes sense with the retag scenario. Since this uses a tag archive as `stable`, we should add an explanatory comment before the `livecheck` block to make sure the situation is clear (otherwise someone may see it and think it's an unnecessary use of `GithubLatest` and naively switch it back). I pushed a commit to add a comment.

[Just to be clear, we only need an explanatory comment when `GithubLatest` is used and the `stable` URL isn't a GitHub release asset. When `stable` is a GitHub release asset, it's readily apparent why we're using `GithubLatest`.]

**Review by chenrui333 — APPROVED** [review]
